did/model.go

```diff
@@ -12,6 +12,9 @@ const (
 	KnownDIDContext string = "https://www.w3.org/ns/did/v1"
 )
 
+type DIDDocumentMetadata map[string]interface{}
+type DIDResolutionMetadata map[string]interface{}
```

Comment:
should we add a few known properties for resolution and document metadata here? 
https://www.w3.org/TR/did-core/#did-resolution-metadata
https://www.w3.org/TR/did-core/#did-document-metadata

this could either be implemented as a few `const`s for each known property, and keep it as a `map[string]interface{}` or create a known object model for the properties we are aware of.